src/server/pfs/server/api_server.go

```diff
@@ -507,11 +575,52 @@ func (a *apiServer) DeleteFile(ctx context.Context, request *pfsclient.DeleteFil
 	a.versionLock.RLock()
 	defer a.versionLock.RUnlock()
 	ctx = versionToContext(a.version, ctx)
-	clientConn, err := a.getClientConnForFile(request.File, a.version)
+
+	fileInfo, err := a.InspectFile(ctx, &pfsclient.InspectFileRequest{
+		File: request.File,
+	})
 	if err != nil {
 		return nil, err
 	}
-	return pfsclient.NewInternalAPIClient(clientConn).DeleteFile(ctx, request)
+
+	if fileInfo.FileType == pfsclient.FileType_FILE_TYPE_REGULAR {
+		clientConn, err := a.getClientConnForFile(request.File, a.version)
+		if err != nil {
+			return nil, err
+		}
+		return pfsclient.NewInternalAPIClient(clientConn).DeleteFile(ctx, request)
+	}
+
+	// If we are deleting a directory, we multicast it to the entire cluster
+	clientConns, err := a.router.GetAllClientConns(a.version)
+	if err != nil {
+		return nil, err
+	}
+	var wg sync.WaitGroup
+	errCh := make(chan error, 1)
+	for _, clientConn := range clientConns {
+		wg.Add(1)
```

Comment:
could you do `clientConn := clientConn` here, I like keeping that standard across the code base.